src/codecs/framing/character_delimited.rs

```diff
@@ -16,7 +16,7 @@ pub struct CharacterDelimitedDecoderConfig {
 #[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
 pub struct CharacterDelimitedDecoderOptions {
     /// The character that delimits byte sequences.
-    delimiter: char,
+    delimiter: u8,
```

Comment:
Unfortunately this subtly lead to a bug: it breaks deserializing chars from the config, as the common formats don't have any single-character type and they will not try to coerce a string -> ascii character -> `u8`.

Reply:
Ah, hmm, shoot. Seems like the thing to do would flip this back to `char` and instead of `memchr::memchr` write our own variant that takes a `char` instead of a byte.

Reply:
I got a branch with a custom `serde` deserializer/serializer for handling `u8` fields as `char`. Keeping the restriction of 1 byte on the encoded field type makes sense to me.

Reply:
PR: #10829

Reply:
Another thought an a high level: I'm considering that we might want to allow strings as a delimiter sooner than later. Consider `\r\n` which should be a fairly common case on Windows. Depending on the performance implications we could special case that to a single character delimiter if applicable, though.